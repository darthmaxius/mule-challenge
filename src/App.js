import React, { useEffect, useState } from "react";
import ReactFlow from "react-flow-renderer";
import TextArea from "./components/textarea";
import Button from "./components/button";

export default function App() {
  const elementsRaw = [
    {
      id: "1",
      type: "input", // input node
      data: { label: "Input Node" },
      position: { x: 250, y: 25 },
    },
    // default node
    {
      id: "2",
      // you can also pass a React component as a label
      data: { label: <div>Default Node</div> },
      position: { x: 100, y: 125 },
    },
    {
      id: "3",
      type: "output", // output node
      data: { label: "Output Node" },
      position: { x: 250, y: 250 },
    },
    // animated edge
    { id: "e1-2", source: "1", target: "2", arrowHeadType: "arrowclosed" },
    { id: "e2-3", source: "2", target: "3", arrowHeadType: "arrowclosed" },
  ];

  const [elements, setElements] = useState([]);

  useEffect(() => {
    setElements(elementsRaw);
  }, []);

  return (
    <div>
      <TextArea hanlderApp={setElements} />
      <Button>Ajustar</Button>
      <div style={{ height: "1000px" }}>
        <ReactFlow elements={elements} />
      </div>
    </div>
  );
}
