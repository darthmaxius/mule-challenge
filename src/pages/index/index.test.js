import { render, fireEvent, screen } from '@testing-library/react';
import App from '.';

jest.mock('react-flow-renderer', () => () => <div />);

describe('Index', () => {
  const setup = () => {
    const utils = render(<App />);
    const textarea = utils.getByLabelText('panel-textarea');
    const button = utils.getByLabelText('panel-button');

    return {
      textarea,
      button,
      ...utils,
    };
  };
  it('Happy render', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Change with valid value in textarea and click in Draw', () => {
    const { textarea, button } = setup();
    fireEvent.change(textarea, { target: { value: 'a depends b' } });

    fireEvent.click(button);

    expect(screen.getByText('Summary')).toBeTruthy();
  });

  it('Change with invalid values [THIS LINE ONLY HAVE ONE DEPENDS] in textarea and click in Draw', () => {
    const { textarea, button } = setup();
    fireEvent.change(textarea, { target: { value: 'a b' } });

    expect(textarea.value).toBe('a b');

    fireEvent.click(button);

    expect(screen.getByText('Summary')).toBeTruthy();
  });
  it('Change with invalid values [INVALID LINE TO DRAW] in textarea and click in Draw', () => {
    const { textarea, button } = setup();
    fireEvent.change(textarea, { target: { value: 'depends' } });

    fireEvent.click(button);

    expect(screen.getByText('Summary')).toBeTruthy();
  });
});
