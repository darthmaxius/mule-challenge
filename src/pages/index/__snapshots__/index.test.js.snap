// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Happy render 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="layout-panel"
    >
      <div
        class="layout-panel__inputs"
      >
        <textarea
          aria-label="panel-textarea"
          class="layout-panel__inputs layout-panel__inputs-textarea"
        />
        <button
          aria-label="panel-button"
          class="layout-panel__inputs layout-panel__inputs-button"
        >
          Draw
        </button>
      </div>
    </div>
    <div
      class="layout-graph"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
